### PR TITLE
Updated spack-deployment workflow to be self contained

### PR DESCRIPTION
**Pull request by LiamBindle**
@laestrada I was reviewing your PR and I was curious if we could keep it self contained in the workflow config file. I was able to do it by moving the set up, authentication, and git commands in `spack-deployment/*` to steps in the workflow, and then the entire workflow is executed in `continuumio/miniconda3`. I think that having this workflow as a standalone YAML file will be easier to maintain in the long run.

**Outstanding issues:**
This sed expression isn't working for me
```
sed -i "1,/version(.*/s//${VERSION_STRING}\n    &/" var/spack/repos/builtin/packages/gchp/package.py
```
Any ideas?

### CODE REVIEW

**Comment by LiamBindle** [comment]
It looks like the PR is failing because I still have `branch` as one of the triggers. After the outstanding issue is resolved we can replace the triggers and resolve the outstanding todos.

**Comment by laestrada** [comment]
> 

@LiamBindle I think this is a great idea -- thanks for trying this out! It took me embarrassingly long to get that sed command to work. I think there are a bunch of slight syntax variations depending on the os system and flavor. I would try using the alpine linux variation of this container `continuumio/miniconda3:4.10.3-alpine`. I was running into some issues with spack when using the untagged container version, so it could be the sed command behaves slightly differently in that container.

File: .github/workflows/spack-deployment-workflow.yml [hunk 1]
```diff
@@ -2,39 +2,80 @@
...
+          # Create a new branch for current release
+          git checkout -b gchp-${GEOSCHEM_VERSION}
+
+          # replace the tarball link to new version
+          sed -i "s|https://github.com/geoschem/GCHP/archive/.*.gz|${TAR_URL_PREFIX}/${GEOSCHEM_VERSION}.tar.gz|" var/spack/repos/builtin/packages/gchp/package.py
+          # add new line to version history
+          # todo: uncomment (needs fix) | sed -i "1,/version(.*/s//${VERSION_STRING}\n    &/" var/spack/repos/builtin/packages/gchp/package.py
+
+          # test that style is up to snuff
+          spack style
```
Comment (laestrada) [inline .github/workflows/spack-deployment-workflow.yml:73]: I was having issues with spack if I didn't install a bunch of dependencies for spack. I'm impressed that spack style is working correctly. Did you run into any errors when installing spack?

**Comment by LiamBindle** [comment]
Sounds good, I'll have a crack at the sed expression. After that I'll merge the PRs. Do you mind maintaining this pipeline still? (i.e., your credentials in 'Login with PAT', and dealing with any issues that come up)

File: .github/workflows/spack-deployment-workflow.yml (hunk 1 above)
Comment (LiamBindle) [reply .github/workflows/spack-deployment-workflow.yml:73]: I didn't but I was having trouble with the alpine tag, so maybe it's a bit unstable. I had to install `isort` but that was the only extra I needed.

**Comment by laestrada** [comment]
Yup, I am fine with that.

**Pull request merged**